### PR TITLE
Frontside navigation

### PR DESCRIPTION
**Pull request by sk8mate**

### CODE REVIEW

File: frontside/lib/main.dart [hunk 1]
```diff
@@ -16,8 +16,9 @@ class Sk8 extends StatelessWidget {
       theme: defaultTheme,
       routeInformationParser: BeamerParser(),
       routerDelegate: BeamerRouter.routerDelegate,
-      backButtonDispatcher:
-          BeamerBackButtonDispatcher(delegate: BeamerRouter.routerDelegate),
+      //https://pub.dev/documentation/beamer/latest/beamer/BeamerBackButtonDispatcher-class.html
+      //Why not stay in default button behavior?
+      backButtonDispatcher: BeamerBackButtonDispatcher(delegate: BeamerRouter.routerDelegate),
```
Comment (sk8mate) [inline frontside/lib/main.dart:21]: I think it's required to provide the same behavior to the Android back button:
> NOTE that Beamer can integrate Android's back button to do beamBack if possible when all the pages from current BeamLocation have been popped. This is achieved by setting a back button dispatcher in MaterialApp.router.

Source: https://pub.dev/packages/beamer#beaming-back

File: frontside/pubspec.lock [hunk 2]
```diff
@@ -153,7 +153,7 @@ packages:
       name: test_api
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "0.4.2"
+    version: "0.3.0"
   typed_data:
     dependency: transitive
     description:
```
Comment (sk8mate) [inline frontside/pubspec.lock:159]: I'm not sure what you did but we don't want to downgrade any dependencies. Can you revert this? :)

File: frontside/lib/views/home/home.dart [hunk 3]
```diff
@@ -14,30 +13,44 @@ class Home extends StatelessWidget {
...
+      ],
+      currentIndex: _seletedItem,
+      onTap: (index) {
+        setState(() {
+          _seletedItem = index;
+        });
+      },
     );
   }
 }
```
[the remark this answers is not in the document]
Comment (Japczy) [reply frontside/lib/views/home/home.dart:56]: 1. Sure
2. We can use 'NavBar' name, it's not reserved.

File: frontside/pubspec.lock (hunk 2 above)
Comment (Japczy) [reply frontside/pubspec.lock:159]: https://dart.dev/guides/libraries/private-files

**Comment by Japczy** [comment]
Idk why after edit gitigore in commit 620a12f git added pubspec file. Any idea?
Finaly it's good cuz probably now tests pass.

File: frontside/pubspec.lock (hunk 2 above)
Comment (sk8mate) [reply frontside/pubspec.lock]: According to the docs we should keep it:
> For application packages, we recommend that you commit the pubspec.lock file. Saving pubspec.lock ensures that everyone working on the app uses the exact same versions.

Source: https://dart.dev/guides/libraries/private-files#pubspeclock

We're developing an application, not a library :)

**Comment by sk8mate** [comment]
> Idk why after edit gitigore in commit [620a12f](https://github.com/sk8mate/sk8/commit/620a12f840281a9d55b928c38f39e0b04d1ede43) git added pubspec file. Any idea?
> Finaly it's good cuz probably now tests pass.

https://github.com/sk8mate/sk8/pull/45#discussion_r706878913

File: frontside/lib/views/home/home.dart [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import 'package:flutter/material.dart';
-import 'package:beamer/beamer.dart';
-import 'package:frontside/views/about/about.dart';
+// import 'package:beamer/beamer.dart';
```
Comment (sk8mate) [inline frontside/lib/views/home/home.dart:2]:
```suggestion
```

**Pull request merged**